src/providers/aws.rs

```diff
@@ -109,6 +110,38 @@ use std::time;
 use tracing::instrument;
 use tracing_futures::Instrument;
 
+/// Dictate how a set of instances should be launched.
+#[derive(Debug, Clone)]
+#[allow(missing_copy_implementations)]
+#[non_exhaustive]
+pub enum LaunchMode {
+    /// Use AWS [defined duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances) spot instances.
+    DefinedDuration {
+        /// The lifetime of the defined duration instances.
+        /// This value must be between 1 and 6 hours.
+        hours: usize,
+    },
+    /// Use regular AWS on-demand instances.
+    OnDemand,
+}
+
+impl LaunchMode {
+    /// Launch using AWS [defined duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances) spot instances.
+    ///
+    /// The lifetime of such instances must be declared in advance (1-6 hours).
+    /// This method thus clamps `hours` to be between 1 and 6.
```

Comment:
This is definitely personal preference but I'd rather barf on invalid input than silently succeed. Totally fine either way as long as it's documented

Reply:
Waiting on those [range types](https://github.com/rust-lang/rfcs/issues/671) I guess :)

Reply:
Yeah, this one was a tricky decision. The main issue I have here is that panicking would mean that cleanup isn't run, which means that _past_ instances may be left running. We could define an `enum` I suppose, but that seems like overkill. My inclination is to keep it as-is.